healthcheck/healthcheck.go

```diff
@@ -127,11 +130,31 @@ func checkBackend(currentBackend *BackendHealthCheck) {
 	}
 }
 
+func (backend *BackendHealthCheck) newRequest(serverURL *url.URL) (*http.Request, error) {
+	if backend.Options.Port == 0 {
+		return http.NewRequest("GET", serverURL.String()+backend.Path, nil)
+	}
+
+	// copy the url and add the port to the host
+	u := &url.URL{}
+	*u = *serverURL
+	u.Host = net.JoinHostPort(u.Hostname(), strconv.Itoa(backend.Options.Port))
+	u.Path = u.Path + backend.Path
+
+	return http.NewRequest("GET", u.String(), nil)
+}
+
 func checkHealth(serverURL *url.URL, backend *BackendHealthCheck) bool {
 	client := http.Client{
 		Timeout: backend.requestTimeout,
 	}
-	resp, err := client.Get(serverURL.String() + backend.Path)
+	req, err := backend.newRequest(serverURL)
+	if err != nil {
+		return false
```

Comment:
Could you please add a log message here. I'm inclined to say it should be of level `ERROR` since we should apparently only run into this case if we have a logic error in our code.